cli/src/cmd/forge/verify.rs

```diff
@@ -235,6 +306,93 @@ To skip this solc dry, pass `--force`.
 
         Ok(())
     }
+
+    fn flattened_source(
+        &self,
+        project: &Project,
+        target: &Path,
+        version: &Version,
+    ) -> eyre::Result<(String, String, CodeFormat)> {
+        let bch = project
+            .solc_config
+            .settings
+            .metadata
+            .as_ref()
+            .and_then(|m| m.bytecode_hash)
+            .unwrap_or_default();
+
+        eyre::ensure!(
+            bch == BytecodeHash::Ipfs,
+            "When using flattened source, bytecodeHash must be set to ipfs. BytecodeHash is currently: {}. Hint: Set the bytecodeHash key in your foundry.toml :)",
```

Comment:
```suggestion
            "When using flattened source, bytecodeHash must be set to ipfs because Etherscan uses IPFS in its Compiler Settings when re-compiling your code. BytecodeHash is currently: {}. Hint: Set the bytecodeHash key in your foundry.toml :)",
```